src/components/PopupWithForm.js

```diff
@@ -25,12 +32,27 @@ export default class PopupWithForm extends Popup{
         });
     }
 
+    //create a renderLoading method - see reviewer recommendation
+    renderLoading(isLoading, loadingText='Saving...') {
+        if(isLoading) {
+            this._submitBtn.textContent = loadingText;
+        } else {
+            this._submitBtn.textContent = this._submitBtnText;
+        }
+    }
+
+    _handleImageSubmit(){ //delete this
+        evt.preventDefault();
+        this._modalImage.classList.remove('modal_opened');
+        this._modalImageSubmit.removeEventListener('submit', this._imageSubmitHandler);
```

Comment:
* There should **not** be any code in universal classes that doesn’t suit every popup.
* Please, delete the hardcode for the avatar from `PopupWithForm`